src/status_im/chat/models/message.cljs

```diff
@@ -151,158 +191,112 @@
+                       (chat-model/upsert-chat {:chat-id chat-id})
+                       (add-message chat-id message-with-id true)
+                       (send chat-id send-record))))
 
 (defn send-message [{:keys [db now random-id] :as cofx} {:keys [chat-id] :as params}]
-  (let [chat    (get-in db [:chats chat-id])
-        message (prepare-message params chat now random-id)
-        params' (assoc params :message message)
-        fx      (-> (chat-model/upsert-chat cofx {:chat-id chat-id})
-                    (update :db add-message-to-db chat-id message true)
-                    (assoc :save-message message))]
-    (merge fx (send cofx params'))))
-
-(defn- prepare-command
-  [identity chat-id now last-from-clock-value last-to-clock-value
+  (upsert-and-send (prepare-plain-message params (get-in db [:chats chat-id]) now) cofx))
+
+(defn- prepare-command-message
+  [identity {:keys [last-to-clock-value last-from-clock-value chat-id] :as chat} now
    {request-params  :params
     request-command :command
     :keys           [prefill prefillBotDb]
     :as             request}
-   {:keys [id params command to-message handler-data content-type]}]
+   {:keys [params command handler-data content-type]}]
   (let [content (if request
-                  {:command        request-command
-                   :params         (assoc request-params :bot-db (:bot-db params))
-                   :prefill        prefill
-                   :prefill-bot-db prefillBotDb}
-                  {:command (:name command)
-                   :scope   (:scope command)
-                   :params  params})
-        content' (assoc content :handler-data handler-data
-                        :type (name (:type command))
-                        :content-command (:name command)
-                        :content-command-scope-bitmask (:scope-bitmask command)
-                        :content-command-ref (:ref command)
-                        :preview (:preview command)
-                        :short-preview (:short-preview command)
-                        :bot (or (:bot command)
-                                 (:owner-id command)))]
-    {:message-id       id
-     :from             identity
-     :to               chat-id
-     :timestamp        now
-     :content          content'
-     :content-type     (or content-type
-                           (if request
-                             constants/content-type-command-request
-                             constants/content-type-command))
-     :outgoing         true
-     :to-message       to-message
-     :type             (:type command)
-     :has-handler      (:has-handler command)
-     :to-clock-value   (inc last-to-clock-value)
-     :from-clock-value last-from-clock-value
-     :show?            true}))
+                  {:request-command     request-command
+                   :request-command-ref (:ref command) ;; TODO janherich this is technically not correct, but works for now
```

Comment:
is this comment related to the current or the next line? Let's make comments above the code, otherwise it's a bit confusing